test/upgrades/RibbonThetaVault.ts

```diff
@@ -75,161 +55,63 @@ describe("RibbonThetaVault upgrade", () => {
-  });
+  checkIfStorageNotCorrupted("0x25751853Eab4D0eB3652B5eB6ecB102A2789644B");
+  checkIfStorageNotCorrupted("0x65a833afDc250D9d38f8CD9bC2B1E3132dB13B2F");
+  checkIfStorageNotCorrupted("0xe63151A0Ed4e5fafdc951D877102cf0977Abd365");
+  checkIfStorageNotCorrupted("0x53773E034d9784153471813dacAFF53dBBB78E8c");
 });
 
-function checkIfStorageNotCorrupted(
-  vaultAddress: string,
-  storageValues: StorageValues
-) {
-  const {
-    totalSupply,
-    name,
-    symbol,
-    vaultParams1,
-    vaultParams2,
-    cap,
-    vaultState1,
-    vaultState2,
-    optionState,
-    feeRecipient,
-    keeper,
-    performanceFee,
-    managementFee,
-    optionsPremiumPricer,
-    strikeSelection,
-    premiumDiscount,
-    currentOtokenPremium,
-    lastStrikeOverrideRound,
-    overriddenStrikePrice,
-    auctionDuration,
-    optionAuctionID,
-  } = storageValues;
-
+function checkIfStorageNotCorrupted(vaultAddress: string) {
   const getVaultStorage = async (storageIndex: number | string) => {
     return await ethers.provider.getStorageAt(vaultAddress, storageIndex);
   };
 
-  const storageLayout = [
-    [
-      ADMIN_SLOT,
-      "0x000000000000000000000000223d59fa315d7693df4238d1a5748c964e615923",
-    ],
-    [0, "0x0000000000000000000000000000000000000000000000000000000000000001"], // initializable
-    [1, "0x0000000000000000000000000000000000000000000000000000000000000001"], // reentrancy
-    [101, "0x00000000000000000000000077da011d5314d80be59e939c2f7ec2f702e1dcc4"], // owner
-    [153, totalSupply], // totalSupply
-    [154, name], // name
-    [155, symbol], // symbol
-    [204, vaultParams1],
-    [205, vaultParams2],
-    [206, cap],
-    [207, vaultState1], // lastLockedAmount + lockedAmount + round
-    [208, vaultState2], // queuedWithdrawShares + totalPending
-    [209, "0x0000000000000000000000000000000000000000000000000000000000000000"],
-    [210, optionState], // nextOption + nextOptionReadyAt + currentOption
-    [211, feeRecipient], // feeRecipient
-    [212, keeper], // keeper
-    [213, performanceFee], // performanceFee
-    [214, managementFee], // managementFee
-    [245, optionsPremiumPricer], // optionsPremiumPricer
-    [246, strikeSelection], // strikeSelection
-    [247, premiumDiscount], // premiumDiscount
-    [248, currentOtokenPremium], // currentOtokenPremium
-    [249, lastStrikeOverrideRound], // lastStrikeOverrideRound
-    [250, overriddenStrikePrice], // overriddenStrikePrice
-    [251, auctionDuration], // auctionDuration
-    [252, optionAuctionID], // optionAuctionID
+  const storageSlots = [
+    ADMIN_SLOT,
+    0,
+    1,
+    101,
+    153,
+    154,
+    155,
+    204,
+    205,
+    206,
+    207,
+    208,
+    209,
+    210,
+    211,
+    212,
+    213,
+    214,
+    245,
+    246,
+    247,
+    248,
+    249,
+    250,
+    251,
+    252,
+    253,
+    254,
+    255,
+    256,
   ];
+  let storageLayout: [number | string, string][];
```

Comment:
Please avoid union types unless you have no other option, they cause more confusion than solve.